### PR TITLE
NEXUS-4574 - Move setProxyBlocking and setOutOfService to RepositoryMessageUtil

### PR DESCRIPTION
**Pull request by velo**
Still looking if I can caught the IT problem, but for now this is a reasonable improvement on current code.
https://builds.sonatype.org/view/nexus/job/nexus-4574/

### CODE REVIEW

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java [hunk 1]
```diff
@@ -505,4 +508,81 @@ public class RepositoryMessageUtil
...
+            status.setRemoteStatus( RemoteStatus.UNAVAILABLE.name() );
+            status.setProxyMode( ProxyMode.ALLOW.name() );
+        }
+        Response response = RepositoryStatusMessageUtil.changeStatus( status );
+
+        if ( !response.getStatus().isSuccess() )
+        {
+            Assert.fail( "Could not unblock proxy: " + repoId + ", status: " + response.getStatus().getName() + " ("
+                + response.getStatus().getCode() + ") - " + response.getStatus().getDescription() );
+        }
```
Comment (peterlynch) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java:550]: Use RequestFacade.\* and NexusRequestMatchers/ResponseMatchers for the above few lines. Make sure Response is released.

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java [hunk 2]
```diff
@@ -505,4 +508,81 @@ public class RepositoryMessageUtil
...
+        {
+            status.setLocalStatus( LocalStatus.IN_SERVICE.name() );
+        }
+        Response response = RepositoryStatusMessageUtil.changeStatus( status );
+
+        if ( !response.getStatus().isSuccess() )
+        {
+            Assert.fail( "Could not set proxy out of service status (Status: " + response.getStatus() + ": " + repoId
+                + "\n" + response.getEntity().getText() );
+        }
```
Comment (peterlynch) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java:590]: same - cleanup Assertions and Response

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java [hunk 3]
```diff
@@ -505,4 +508,81 @@ public class RepositoryMessageUtil
...
+
+    /**
+     * Change block proxy state.<BR>
+     * this method only return after all Tasks and Asynchronous events to finish
+     * 
+     * @param repoId
+     * @param block
+     * @throws Exception
+     */
+    public void setBlockProxy( String repoId, boolean block )
```
Comment (peterlynch) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java]: not a blocker but good practice imo to use final on the arguments, especially primitives

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/integrationtests/AbstractNexusProxyIntegrationTest.java [hunk 4]
```diff
@@ -94,58 +100,6 @@ public abstract class AbstractNexusProxyIntegrationTest
...
-        {
-            status.setRemoteStatus( RemoteStatus.AVAILABLE.name() );
-            status.setProxyMode( ProxyMode.BLOCKED_MANUAL.name() );
-        }
-        else
-        {
-            status.setRemoteStatus( RemoteStatus.UNAVAILABLE.name() );
-            status.setProxyMode( ProxyMode.ALLOW.name() );
-        }
-        Response response = RepositoryStatusMessageUtil.changeStatus( status );
```
Comment (bdemers) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/integrationtests/AbstractNexusProxyIntegrationTest.java:116]: -don't forget to release the response-
sorry, my eyes are missing the red.... disregard....

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java [hunk 5]
```diff
@@ -505,4 +510,75 @@ public class RepositoryMessageUtil
...
+        {
+            status.setRemoteStatus( RemoteStatus.AVAILABLE.name() );
+            status.setProxyMode( ProxyMode.BLOCKED_MANUAL.name() );
+        }
+        else
+        {
+            status.setRemoteStatus( RemoteStatus.UNAVAILABLE.name() );
+            status.setProxyMode( ProxyMode.ALLOW.name() );
+        }
+        Response response = RepositoryStatusMessageUtil.changeStatus( status );
```
Comment (bdemers) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java:539]: Release the response

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java [hunk 6]
```diff
@@ -505,4 +510,75 @@ public class RepositoryMessageUtil
...
+        status.setRepoType( "proxy" );
+        if ( outOfService )
+        {
+            status.setLocalStatus( LocalStatus.OUT_OF_SERVICE.name() );
+        }
+        else
+        {
+            status.setLocalStatus( LocalStatus.IN_SERVICE.name() );
+        }
+        Response response = RepositoryStatusMessageUtil.changeStatus( status );
```
Comment (bdemers) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/RepositoryMessageUtil.java:581]: set me free

**Comment by bdemers** [comment]
+1

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/TaskScheduleUtil.java [hunk 7]
```diff
@@ -79,12 +79,18 @@ public class TaskScheduleUtil
         return null;
     }
 
+    /**
+     * @return only tasks visible from nexus UI
+     */
     public static List<ScheduledServiceListResource> getTasks()
```
Comment (peterlynch) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/TaskScheduleUtil.java:85]: thanks for the javadoc

**Comment by peterlynch** [comment]
+1

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/TaskScheduleUtil.java (hunk 7 above)
Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/TaskScheduleUtil.java:85]: Holiday bonus =D

**Pull request merged**